src/Microsoft.ML.TimeSeries/SrCnnTransformBase.cs

```diff
@@ -0,0 +1,319 @@
+    {
+        [Argument(ArgumentType.Required, HelpText = "The name of the source column.", ShortName = "src",
+            SortOrder = 1, Purpose = SpecialPurpose.ColumnName)]
+        public string Source;
+
+        [Argument(ArgumentType.Required, HelpText = "The name of the new column.",
+            SortOrder = 2)]
+        public string Name;
+
+        [Argument(ArgumentType.AtMostOnce, HelpText = "The size of the sliding window for computing spectral residual", ShortName = "wnd",
+            SortOrder = 3)]
+        public int WindowSize = 24;
+
+        [Argument(ArgumentType.AtMostOnce, HelpText = "The size of the initial window for computingd. The default value is set to 0, which means there is no initial window considered.", ShortName = "iwnd",
+            SortOrder = 4)]
+        public int InitialWindowSize = 0;
+
+        [Argument(ArgumentType.AtMostOnce, HelpText = "The number of points to the back of training window.",
+            ShortName = "backwnd", SortOrder = 5)]
+        public int BackAddWindowSize = 5;
+
+        [Argument(ArgumentType.AtMostOnce, HelpText = "The number of pervious points used in prediction.",
+            ShortName = "aheadwnd", SortOrder = 6)]
+        public int LookaheadWindowSize = 5;
+
+        [Argument(ArgumentType.Required, HelpText = "The size of sliding window to generate a saliency map for the series.",
+            ShortName = "avgwnd", SortOrder = 7)]
+        public int AvergingWindowSize = 3;
+
+        [Argument(ArgumentType.Required, HelpText = "The size of sliding window to generate a saliency map for the series.",
+            ShortName = "jdgwnd", SortOrder = 8)]
+        public int JudgementWindowSize = 21;
+
+        [Argument(ArgumentType.AtMostOnce, HelpText = "The threshold to determine anomaly, score larger than the threshold is considered as anomaly.",
+            ShortName = "thre", SortOrder = 9)]
+        public double Threshold = 0.3;
+    }
+
+    internal abstract class SrCnnTransformBase<TInput, TState> : SequentialTransformerBase<TInput, VBuffer<Double>, TState>
+        where TState : SrCnnTransformBase<TInput, TState>.SrCnnStateBase, new()
+    {
+        internal int BackAddWindowSize;
+
+        internal int LookaheadWindowSize;
+
+        internal int AvergingWindowSize;
+
+        internal int JudgementWindowSize;
+
+        internal Double AlertThreshold;
+
+        internal int OutputLength;
+
+        private protected SrCnnTransformBase(int windowSize, int initialWindowSize, string inputColumnName, string outputColumnName, string name, IHostEnvironment env,
+            int backAddWindowSize, int lookaheadWindowSize, int averagingWindowSize, int judgementWindowSize, Double alertThreshold)
+            : base(Contracts.CheckRef(env, nameof(env)).Register(name), windowSize, initialWindowSize, outputColumnName, inputColumnName, new VectorDataViewType(NumberDataViewType.Double, 3))
+        {
+            //TODO: Check argument
```

Comment:
What is the plan for addressing this TODO?

Reply:
Added argument checking for this TODO.